web/src/components/grouped-score-badge.tsx

```diff
@@ -5,88 +5,161 @@ import {
 import {
-  isBooleanDataType,
-  isCategoricalDataType,
-} from "@/src/features/manual-scoring/lib/helpers";
-import { type ScoreSimplified } from "@/src/server/api/routers/generations/getAllQuery";
+  type QuantitativeAggregate,
+  type QualitativeAggregate,
+} from "@/src/features/manual-scoring/lib/aggregateScores";
+import { type APIScore } from "@/src/features/public-api/types/scores";
+
+import { numberFormatter } from "@/src/utils/numbers";
 import { cn } from "@/src/utils/tailwind";
+import { type Score } from "@langfuse/shared";
 import { MessageCircleMore } from "lucide-react";
 
-export const GroupedScoreBadges = ({
-  scores,
-  variant = "badge",
+const COLOR_MAP = new Map([
+  ["True", "bg-light-green p-0.5 text-dark-green"],
+  ["False", "bg-light-red p-0.5 text-dark-red"],
+]);
+
+const SingleScoreValue = ({
+  value,
+  comment,
+  showColorCoding = false,
 }: {
-  scores: ScoreSimplified[];
-  variant?: "badge" | "headings";
+  value: string;
+  comment?: string | null;
+  showColorCoding?: boolean;
 }) => {
-  const groupedScores = scores.reduce<Record<string, ScoreSimplified[]>>(
-    (acc, score) => {
-      if (!acc[score.name] || !Array.isArray(acc[score.name])) {
-        acc[score.name] = [score];
-      } else {
-        (acc[score.name] as ScoreSimplified[]).push(score);
-      }
-      return acc;
-    },
-    {},
+  return (
+    <span
+      className={cn(
+        "group/score ml-1 rounded-sm first:ml-0",
+        showColorCoding && COLOR_MAP.get(value),
+      )}
+    >
+      {value}
+      {comment && (
+        <HoverCard>
+          <HoverCardTrigger className="ml-1 inline-block cursor-pointer">
+            <MessageCircleMore size={12} />
+          </HoverCardTrigger>
+          <HoverCardContent className="overflow-hidden whitespace-normal break-normal">
+            <p>{comment}</p>
+          </HoverCardContent>
+        </HoverCard>
+      )}
+      <span className="group-last/score:hidden">,</span>
+    </span>
   );
+};
 
-  const ScoresOfGroup = (props: {
-    scores: ScoreSimplified[];
-    className?: string;
-  }) => (
-    <div className={cn("text-xs", props.className)}>
-      {props.scores.map((s, i) => (
-        <span key={i} className="group/score ml-1 first:ml-0">
-          {isCategoricalDataType(s.dataType) || isBooleanDataType(s.dataType)
-            ? s.stringValue
-            : s.value?.toFixed(2)}
-          {s.comment && (
-            <HoverCard>
-              <HoverCardTrigger className="ml-1 inline-block cursor-pointer">
-                <MessageCircleMore size={12} />
-              </HoverCardTrigger>
-              <HoverCardContent className="overflow-hidden whitespace-normal break-normal">
-                <p>{s.comment}</p>
-              </HoverCardContent>
-            </HoverCard>
-          )}
-          <span className="group-last/score:hidden">,</span>
-        </span>
-      ))}
-    </div>
-  );
+const DETAIL_HOVER_AFTER = 2;
```

Comment:
i thought this was a delay for a hover and then realized it is the number after which categories are truncated. I'd rename to sth like `COLLAPSE_CATEGORICAL_SCORES_AFTER=2`